### PR TITLE
Turn on testing for windows

### PR DESCRIPTION
**Pull request by achew22**

### CODE REVIEW

File: tools/repositories.bzl [hunk 1]
```diff
@@ -53,9 +89,10 @@ Args:
 Limitation: only support Linux and macOS for now.
 """
 
-def bazel_binaries(versions = BAZEL_VERSIONS):
+
+def bazel_binaries(versions=BAZEL_VERSIONS):
```
Comment (ittaiz) [inline tools/repositories.bzl]: Did you remove the spaces here and below on purpose?

File: tools/repositories.bzl [hunk 2]
```diff
@@ -14,37 +14,73 @@
...
+
+  url = _BAZEL_BINARY_PACKAGE.format(version=version, platform=platform, extension=extension)
+  args = {"url": url, "type": "zip", "output": "bin"}
   if version in BAZEL_HASH_DICT and platform in BAZEL_HASH_DICT[version]:
     args["sha256"] = BAZEL_HASH_DICT[version][platform]
   rctx.download_and_extract(**args)
 
+
+def _extract_bazel(rctx):
+  result = rctx.execute([
```
Comment (ittaiz) [inline tools/repositories.bzl]: Extracting Bazel breaks remote execution. Any functional motivation to do this or just performance?

**Comment by achew22** [comment]
I was working with another Googler trying to figure out if we could get this test working in Windows. If you don't mind leaving this open for a bit we discovered a bug in Bazel itself, and 1 enhancement that was necessary in this repo, 2 bugs in iBazel and maybe 1 in the windows CI before getting stuck and not being able to continue. We are waiting for another person to come back from vacation then we will get back on it.

**Comment by ittaiz** [comment]
Sure thing. No rush at all.
Ping me when this stabilizes...
But please keep in mind that extraction and RBE currently don’t mix

File: tools/repositories.bzl [hunk 3]
```diff
@@ -14,37 +14,73 @@
 
 load(":common.bzl", "BAZEL_HASH_DICT", "BAZEL_VERSIONS")
 
-_BAZEL_BINARY_PACKAGE = "http://releases.bazel.build/{version}/release/bazel-{version}-installer-{platform}.sh"
+_BAZEL_BINARY_PACKAGE = "http://releases.bazel.build/{version}/release/bazel-{version}-without-jdk-installer-{platform}.{extension}"
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply tools/repositories.bzl]: Done

File: .bazelci/presubmit.yml [hunk 4]
```diff
@@ -21,3 +21,13 @@ platforms:
     - "--test_output=all"
     test_targets:
     - "..."
+  windows:
+    build_flags:
+    - "--build_tag_filters=-nowindows"
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply .bazelci/presubmit.yml]: It will filter out all tests tagged with `nowindows`, so that they won't run on Windows.

**Comment by meteorcloudy** [comment]
The CI issue has been resolved. We can test this change properly now.

**Comment by jchv** [comment]
I did a bit of work on getting Windows working a while ago, but never ended up pushing it anywhere. I made some changes to the Python script so it generated both the URL and the hash, because the Windows URL is annoyingly different from the rest and I didn't want to duplicate the URLs even more (both the Starlark side and the Python side generate URLs as of now.) Also, the old version of rules_go needs to be updated to work on Windows (#95.)

Should I proceed on a separate PR with some of those changes?

**Comment by ittaiz** [comment]
Is this still alive? and how does this relate to #96 ?

**Comment by meteorcloudy** [comment]
Closing due to https://github.com/bazelbuild/bazel-integration-testing/pull/117

**Pull request closed without merging**